tetraelf-hierarchy/hierarchy-tool/tool/src/java/org/sakaiproject/hierarchy/tool/entityproviders/HierarchyEntityProviderImpl.java

```diff
@@ -98,4 +98,47 @@ else if(nodeChild instanceof PortalNodeSite){
 		}
 		return hierarchySiteSummary;
 	}
-}
+
+	/**
+	 * /portal-hierarchy/locations.json?siteid=id)
+	 */
+	@EntityCustomAction(action = "locations", viewKey = EntityView.VIEW_LIST)
+	public List<PortalNodeSummary> getNodesFromRootForSite(EntityView view, Map<String, Object> params){
+		String siteId =  (String)params.get("siteid");
+
+		if(StringUtils.isBlank(siteId)){
+			throw new IllegalArgumentException("site ID must be set in order to get the site node locations");
+		}
+
+		List<PortalNodeSummary> portalNodeSummaries = new ArrayList<>();
+
+		if (portalHierarchyService.siteExists(siteId)) {
+
+			List<PortalNode> portalNodes = portalHierarchyService.getNodesWithSite(siteId);
+			PortalNode portalNodePrimary = portalHierarchyService.getDefaultNode(siteId);
+
+			// Site exists but if it is not in the hierarchy show an empty entity.
+			if (portalNodes.isEmpty()) {
+				return portalNodeSummaries;
+			}
+
+			for (PortalNode pn : portalNodes) {
+				if (pn.canView()) {
+					PortalNodeSummary pne = new PortalNodeSummary(pn);
+					if ((pn.getId()).equals(portalNodePrimary.getId())) {
```

Comment:
How about `pn.equals(portalNodePrimary)`?
The equals is `null` safe in the case there isn't a default node (unlikely). When reading code I'm wondering what's the reason for not using the plain `equals()`, eg, what edge case is this trying to catch.